cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/preview/DefaultPreviewManager.java

```diff
@@ -199,7 +201,8 @@ protected void shutDown() throws Exception {
   }
 
   @Override
-  public ApplicationId start(NamespaceId namespace, AppRequest<?> appRequest) throws Exception {
+  @AuthEnforce(entities = "namespaceId", enforceOn = NamespaceId.class, actions = Action.ADMIN)
```

Comment:
As discussed offline, Action.Admin is more like  system/platform read + write permissions

So it is okay to require such Admin permission for preview. In the future, might be better to break this up into separate permissions.

Reply:
Acknowledged. As you stated, READ, WRITE, and EXECUTE are all their own actions, whereas ADMIN currently is more of its own "platform READ/WRITE" privilege rather than a combination of read, write, and execute. We definitely should look into splitting this action up in the future to allow for finer-grain permission checks. I went ahead and modified the Javadoc for Action.ADMIN to more accurately reflect what it does.